src/Storage/Controllers/CleanupController.cs

```diff
@@ -123,11 +123,19 @@ public async Task<ActionResult> CleanupDataelements()
 
             Stopwatch stopwatch = Stopwatch.StartNew();
 
-            foreach (DataElement dataElement in dataElements)
+            Application app = null;
+            Instance instance = null;
+            foreach (DataElement dataElement in dataElements.OrderBy(d => d.InstanceGuid))
             {
                 try
                 {
-                    if (!await blobRepository.DeleteBlob(dataElement.BlobStoragePath.Split('/')[0], dataElement.BlobStoragePath))
+                    if (instance == null || instance.Id != dataElement.InstanceGuid)
```

Comment:
I have a feeling that this is never false because of differences in handling of instance id vs instance guid on the data element. The GetOne method in pgInstanceRepository manipulates the id to include party id.

Reply:
You are right. It's corrected now.